File: Day304_2025/Q304_Leetcode.java
class Solution {
    public int[] getSneakyNumbers(int[] nums) {
        int n = nums.length - 2;
        long sum = 0, squaredSum = 0;

        for (int x : nums) {
            sum += x;
            squaredSum += (long) x * x;
        }

        long expectedSum = (long) n * (n - 1) / 2;
        long expectedSquaredSum = (long) n * (n - 1) * (2L * n - 1) / 6;

        long sum2 = sum - expectedSum;
        long squaredSum2 = squaredSum - expectedSquaredSum;

        double sqrtVal = Math.sqrt(2.0 * squaredSum2 - sum2 * sum2);
        int x1 = (int) ((sum2 - sqrtVal) / 2.0);
        int x2 = (int) ((sum2 + sqrtVal) / 2.0);

        return new int[]{x1, x2};
    }
}

File: Day304_2025/Q304_Leetcode.py
import math
from typing import List

class Solution:
    def getSneakyNumbers(self, nums: List[int]) -> List[int]:
        n = len(nums) - 2
        total_sum = sum(nums)
        squared_sum = sum(x * x for x in nums)

        expected_sum = n * (n - 1) // 2
        expected_squared_sum = n * (n - 1) * (2 * n - 1) // 6

        sum2 = total_sum - expected_sum
        squared_sum2 = squared_sum - expected_squared_sum

        sqrt_val = math.sqrt(2 * squared_sum2 - sum2 * sum2)
        x1 = int((sum2 - sqrt_val) / 2)
        x2 = int((sum2 + sqrt_val) / 2)

        return [x1, x2]
